node/src/service/p2p/bootstrap.rs

```diff
@@ -37,22 +56,47 @@ impl BootstrapConfig {
                 yamux::Config::default,
             )
             .map_err(|_| P2PError::InvalidTcpConfig)?
-            .with_behaviour(|key| BootstrapBehaviour {
-                // Rendezvous server behaviour for serving new peers to connecting nodes.
-                rendezvous: rendezvous::server::Behaviour::new(
-                    rendezvous::server::Config::default().with_max_ttl(DEFAULT_REGISTRATION_TTL), // Max TTL of 24 hours
-                ),
-                // The identify behaviour is used to share the external address and the public key with connecting clients.
-                identify: identify::Behaviour::new(identify::Config::new(
-                    "identify/1.0.0".to_string(),
-                    key.public(),
-                )),
+            .with_behaviour(|key| {
+                // To content-address message, we can take the hash of message and use it as an ID.
+                let message_id_fn = |message: &gossipsub::Message| {
+                    let mut s = DefaultHasher::new();
+                    message.data.hash(&mut s);
+                    gossipsub::MessageId::from(s.finish().to_string())
+                };
+                let gossipsub_config = gossipsub::ConfigBuilder::default()
+                    .message_id_fn(message_id_fn)
+                    .build()
+                    .map_err(|msg| Error::new(ErrorKind::Other, msg))?;
+
+                Ok(BootstrapBehaviour {
+                    // Rendezvous server behaviour for serving new peers to connecting nodes.
+                    rendezvous: rendezvous::server::Behaviour::new(
+                        rendezvous::server::Config::default()
+                            .with_max_ttl(DEFAULT_REGISTRATION_TTL), // Max TTL of 24 hours
+                    ),
+                    // The identify behaviour is used to share the external address and the public key with connecting clients.
+                    identify: identify::Behaviour::new(identify::Config::new(
+                        "identify/1.0.0".to_string(),
```

Comment:
nit: I can see that rendezvous doesn't have this specified and I'd rather have those as constant somewhere.